src/provideEndpoints.ts

```diff
@@ -0,0 +1,69 @@
+import { Endpoints } from './models/core/Endpoints';
+import { Environment } from './models/core/Environment';
+import { CloudRegion } from './models/core/CloudRegion';
+import { ServingConfig } from './models/core/SearchConfig';
+
+export const defaultApiVersion = 20220511;
+
+/**
+ * Provides methods for getting various endpoints.
+ *
+ * @internal
+ */
+export class EndpointsFactory {
+  private readonly environment: Environment;
+  private readonly cloudRegion: CloudRegion;
+
+  constructor(config?: ServingConfig) {
+    this.environment = config?.environment || Environment.PROD;
+    this.cloudRegion = config?.cloudRegion || CloudRegion.US;
+  }
+
+  /** Provides the domain based on environment and cloud region.
+   * @internal
+   */
```

Comment:
nit: I don't believe the internal tags are necessary on the methods because the class is marked as internal, so it is implied that the methods are as well

Reply:
Done